NucleonSecurity/nucleonedr/ingest/parser.yml

```diff
@@ -0,0 +1,128 @@
+    actions:
+      - set:
+          event.category: ["network"]
+          event.type: ["connection"]
+          network.transport: "{{ json.message.protocol }}"
+          source.ip: "{{ json.message.source_ip }}"
+          source.port: "{{ json.message.source_port }}"
+          destination.ip: "{{ json.message.destination_ip }}"
+          destination.port: "{{ json.message.destination_port }}"
+    filter: "{{ json.message.type == 'network' }}"
+
+  file_read:
+    actions:
+      - set:
+          event.category: ["file"]
+          event.type: ["access"]
+          file.path: "{{ json.message.file_path[0] }}"
+          file.size: "{{ json.message.file_size }}"
+          file.hash.md5: "{{ json.message.file_md5 }}"
+          file.hash.sha1: "{{ json.message.file_sha1 }}"
+          file.hash.sha256: "{{ json.message.file_sha256 }}"
+          file.code_signature.subject_name: "{{ json.message.file_signature }}"
+          file.code_signature.valid: "{{ (json.message.file_signed == 2) | default(false) }}"
+    filter: "{{ json.message.type == 'read' }}"
+
+  file_change:
+    actions:
+      - set:
+          event.category: ["file"]
+          event.type: ["change"]
+          file.path: "{{ json.message.file_path[0] }}"
+          file.size: "{{ json.message.file_size }}"
+          file.hash.md5: "{{ json.message.file_md5 }}"
+          file.hash.sha1: "{{ json.message.file_sha1 }}"
+          file.hash.sha256: "{{ json.message.file_sha256 }}"
+          file.code_signature.subject_name: "{{ json.message.file_signature }}"
+          file.code_signature.valid: "{{ (json.message.file_signed == 2) | default(false) }}"
+    filter: "{{ json.message.type in ['write','rename'] }}"
+
+  file_delete:
+    actions:
+      - set:
+          event.category: ["file"]
+          event.type: ["deletion"]
+          file.path: "{{ json.message.file_path[0] }}"
+          file.size: "{{ json.message.file_size }}"
+          file.hash.md5: "{{ json.message.file_md5 }}"
+          file.hash.sha1: "{{ json.message.file_sha1 }}"
+          file.hash.sha256: "{{ json.message.file_sha256 }}"
+          file.code_signature.subject_name: "{{ json.message.file_signature }}"
+          file.code_signature.valid: "{{ (json.message.file_signed == 2) | default(false) }}"
+    filter: "{{ json.message.type == 'delete' }}"
+
+  openproc_stage:
+    actions:
+      - set:
+          event.category: ["process"]
+          event.type: ["info"]
+          file.path: "{{ json.message.file_path[0] }}"
+          file.size: "{{ json.message.file_size }}"
+          file.hash.md5: "{{ json.message.file_md5 }}"
+          file.hash.sha1: "{{ json.message.file_sha1 }}"
+          file.hash.sha256: "{{ json.message.file_sha256 }}"
+          file.code_signature.subject_name: "{{ json.message.file_signature }}"
+          file.code_signature.valid: "{{ (json.message.file_signed == 2) | default(false) }}"
+    filter: "{{ json.message.type == 'open_process' }}"

```

Comment:
bad indentation; it must be at the same level than `set`.
However, for a better readability, I suggest moving the filter in the `pipeline` block.
```suggestion

```